### PR TITLE
feat(graph): add input accordion on project details page

### PR DESCRIPTION
**Pull request by xiongemi**
## Expected Behavior

add an accordion under inputs on the project details page
Screenshot 2024-03-21 at 1 57 46 AM
Screenshot 2024-03-21 at 1 57 38 AM
Screenshot 2024-03-21 at 1 57 31 AM

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/command-line/graph/inputs-utils.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  const externalInputs: string[] = [];
+  const otherInputs: string[] = [];
+  inputs.forEach((input) => {
+    if (input.startsWith('{workspaceRoot}')) {
+      workspaceRootInputs.push(input);
+      return;
+    }
+    if (input.startsWith('{projectRoot}')) {
+      const projectName = taskId.split(':')[0];
+      projectRootInputs.push(`${projectName}:${input}`);
```
Comment (MaxKless) [inline packages/nx/src/command-line/graph/inputs-utils.ts:90]: this is a nit but is it really the best way to add the `projectName` to the string only to replace it again later? Shouldn't we instead use an object or some other data structure vs. a string?
Just a thought, feel free to ignore :P

File: graph/ui-tooltips/src/lib/task-inputs-accordions.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+export function TaskInputsListAccordion({
+  children,
+  heading,
+  containsOnlyOneSection,
+}: {
+  children: ReactNode;
+  heading: string;
+  containsOnlyOneSection: boolean;
```
Comment (MaxKless) [inline graph/ui-tooltips/src/lib/task-inputs-accordions.tsx:122]: feels a little weird that you have to pass this into the component. 
Couldn't `TaskInputsListAccordion` just receive the `projectName` & `inputs` and render the lists itself?
You're way better at react so maybe that's just me not being in tune with the way things are done...

File: e2e/nx-misc/src/extras.test.ts [hunk 3]
```diff
@@ -400,45 +397,42 @@ describe('Extra Nx Misc Tests', () => {
...
-            "libs/lib-dependent-123/tsconfig.json",
-            "libs/lib-dependent-123/tsconfig.lib.json",
-            "libs/lib-dependent-123/tsconfig.spec.json",
-          ],
-        }
-      `);
+      expect(readExpandedTaskInputResponse()[`${baseLib}:build`]).toEqual({
+        '^default': {
+          'lib-dependent-123': {
+            'lib-dependent-123:{projectRoot}/**/*': [
```
Comment (MaxKless) [inline e2e/nx-misc/src/extras.test.ts:403]: why do we need the project name twice here?
Wouldn't this be enough:
```
 'lib-dependent-123': {
      '{projectRoot}/**/*': [
         ...
```

**Review by MaxKless — CHANGES_REQUESTED** [review]
Looks good, still have a few comments & something is broken right now.

File: graph/client-e2e/src/e2e/dev-task-graph.cy.ts [hunk 4]
```diff
@@ -186,52 +173,42 @@ describe('dev mode - task graph', () => {
...
       });
 
       const sharedHeaderSelector =
-        '[data-cy="input-section-entry"]:contains(shared-header)';
-      cy.get(sharedHeaderSelector).click();
-      cy.get(sharedHeaderSelector)
-        .nextAll('[data-cy="input-list-entry"]')
-        .should('have.length', 9);
+        'li[data-cy="input-section-entry"][id="cart-cart-page"]';
+      debugger;
```
[the remark this answers is not in the document]
Comment (xiongemi) [reply graph/client-e2e/src/e2e/dev-task-graph.cy.ts:209]: removed

File: packages/nx/src/command-line/graph/inputs-utils.ts (hunk 1 above)
Comment (xiongemi) [reply packages/nx/src/command-line/graph/inputs-utils.ts]: changed it be an object

File: e2e/nx-misc/src/extras.test.ts (hunk 3 above)
Comment (xiongemi) [reply e2e/nx-misc/src/extras.test.ts]: changed

File: packages/nx/src/command-line/graph/inputs-utils.ts [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    general: Object.values(general).length ? general : undefined,
+    external: Object.values(external).length ? external : undefined,
+    ...project,
+  };
+}
+
+/**
+ * This function is used to categorize the inputs into different categories
+ */
+function categorizeInputs(
```
[the remark this answers is not in the document]
Comment (xiongemi) [reply packages/nx/src/command-line/graph/inputs-utils.ts:70]: added

**Review by MaxKless — APPROVED** [review]

**Review by bcabanes — CHANGES_REQUESTED** [review]
Thank you for the PR, that's a no for me on the UI part. I sent feedback on the slack channel and I'm open for a sync when you like.

**Pull request closed without merging**